instrumentation/jdbc/testing/src/main/java/io/opentelemetry/instrumentation/jdbc/testing/AbstractJdbcInstrumentationTest.java

```diff
@@ -0,0 +1,1918 @@
+    jdbcUserNames.put("h2", null);
+    jdbcUserNames.put("hsqldb", "SA");
+
+    connectionProps.put("databaseName", "someDb");
+    connectionProps.put("OPEN_NEW", "true"); // So H2 doesn't complain about username/password.
+  }
+
+  @BeforeAll
+  static void setUp() {
+    prepareConnectionPoolDatasources();
+  }
+
+  @AfterAll
+  static void tearDown() {
+    cpDatasources
+        .values()
+        .forEach(
+            k ->
+                k.values()
+                    .forEach(
+                        dataSource -> {
+                          if (dataSource instanceof Closeable) {
+                            try {
+                              ((Closeable) dataSource).close();
+                            } catch (IOException ignore) {
+                              // ignore
+                            }
+                          }
+                        }));
+  }
+
+  static void prepareConnectionPoolDatasources() {
+    List<String> connectionPoolNames = asList("tomcat", "hikari", "c3p0");
+    connectionPoolNames.forEach(
+        cpName -> {
+          Map<String, DataSource> dbDsMapping = new HashMap<>();
+          jdbcUrls.forEach(
+              (dbType, jdbcUrl) -> dbDsMapping.put(dbType, createDs(cpName, dbType, jdbcUrl)));
+          cpDatasources.put(cpName, dbDsMapping);
+        });
+  }
+
+  static DataSource createTomcatDs(String dbType, String jdbcUrl) {
+    org.apache.tomcat.jdbc.pool.DataSource ds = new org.apache.tomcat.jdbc.pool.DataSource();
+    String jdbcUrlToSet = dbType.equals("derby") ? jdbcUrl + ";create=true" : jdbcUrl;
+    ds.setUrl(jdbcUrlToSet);
+    ds.setDriverClassName(jdbcDriverClassNames.get(dbType));
+    String username = jdbcUserNames.get(dbType);
+    if (username != null) {
+      ds.setUsername(username);
+    }
+    ds.setPassword("");
+    ds.setMaxActive(1); // to test proper caching, having > 1 max active connection will be hard to
+    // determine whether the connection is properly cached
+    return ds;
+  }
+
+  static DataSource createHikariDs(String dbType, String jdbcUrl) {
+    HikariConfig config = new HikariConfig();
+    String jdbcUrlToSet = dbType.equals("derby") ? jdbcUrl + ";create=true" : jdbcUrl;
+    config.setJdbcUrl(jdbcUrlToSet);
+    String username = jdbcUserNames.get(dbType);
+    if (username != null) {
+      config.setUsername(username);
+    }
+    config.setPassword("");
+    config.addDataSourceProperty("cachePrepStmts", "true");
+    config.addDataSourceProperty("prepStmtCacheSize", "250");
+    config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+    config.setMaximumPoolSize(1);
+
+    return new HikariDataSource(config);
+  }
+
+  static DataSource createC3P0Ds(String dbType, String jdbcUrl) {
+    ComboPooledDataSource ds = new ComboPooledDataSource();
+    try {
+      ds.setDriverClass(jdbcDriverClassNames.get(dbType));
+    } catch (PropertyVetoException e) {
+      throw new IllegalStateException(e);
```

Comment:
this changed from `RuntimeException` to `IllegalStateException` b/c it's now in src/main and so errorprone rules apply